### ISSUE TITLE
[8.76.0] Redis flush array_unique error

### ISSUE THREAD

**Issue by it-can** [laravel/framework#40037]
- Laravel Version: 8.76.0
- PHP Version: 8.0.13
- Database Driver & Version: MariaDB & Redis

### Description:
Flushing redis generates this error:
```
array_unique(): Argument #1 ($array) must be of type array, null given
```
I think it has to do with #39939 

Reverting back to 8.75.0 everything works again.

### Steps To Reproduce:

```
Cache::tags('menu')->flush();
```
```
array_unique(): Argument #1 ($array) must be of type array, null given
```

**Comment by dkuzmenchuk** [comment]
Hi @it-can,

Could you please tell what redis client do you use, predis or phpredis?

**Comment by it-can** [comment]
phpredis

**Comment by DarkPain0** [comment]
We have the same issue after the last upgrade with phpredis

**Comment by taylorotwell** [comment]
@dkuzmenchuk do you understand the problem and are you able to fix it? Otherwise, we need to revert this ASAP.

**Comment by dkuzmenchuk** [comment]
@taylorotwell yes, I found a reason of this issue. Making new PR with a fix right now

**Comment by taylorotwell** [comment]
@dkuzmenchuk do you have a time frame in mind for that?

**Comment by dkuzmenchuk** [comment]
@taylorotwell I think there are more differences between Redis clients than just the return value. So I think we should revert my PR. In this case I'll have time to make more tests to be sure everything work good with my fixes in new PR

**Comment by taylorotwell** [comment]
Reverted.

**Comment by dkuzmenchuk** [comment]
I published new PR #40039 which contains changes from PR #39939, but takes into consideration that Laravel supports 2 Redis clients
